libbeat/beat/info.go

```diff
@@ -21,12 +21,13 @@ import "github.com/gofrs/uuid"
 
 // Info stores a beats instance meta data.
 type Info struct {
-	Beat        string    // The actual beat's name
-	IndexPrefix string    // The beat's index prefix in Elasticsearch.
-	Version     string    // The beat version. Defaults to the libbeat version when an implementation does not set a version
-	Name        string    // configured beat name
-	Hostname    string    // hostname
-	UUID        uuid.UUID // ID assigned to beat instance
+	Beat          string    // The actual beat's name
+	IndexPrefix   string    // The beat's index prefix in Elasticsearch.
+	Version       string    // The beat version. Defaults to the libbeat version when an implementation does not set a version
+	Name          string    // configured beat name
+	Hostname      string    // hostname
+	UUID          uuid.UUID // ID assigned to beat instance
+	EphemeralUUID uuid.UUID // ID assigned to beat instance
```

Comment:
We already have a ephemeral id: https://github.com/elastic/beats/blob/master/libbeat/cmd/instance/metrics_common.go#L31

Yes, not easy to find. I think here is a better place for it.

Reply:
Errr, this isn't a new UUID, we pull it from https://github.com/elastic/beats/pull/9404/files#diff-73bd591578392eb2d281f4778ac9a47eR88

Reply:
Oops, I meant to say: https://github.com/elastic/beats/pull/9404/files#diff-f7226280e7734d6e3d88e7dde232602eR217

Reply:
Actually, it already is pulled from where you mentioned (this line is the type declaration)